### PR TITLE
TRW-1440 - seperate cell structure from row component in table

### PR DESCRIPTION
**Pull request by troyblank**
This is a small step to get the context menu working for cells in a table. Eventually the `Cell` component's state will control the menu visually so the user can select the copy functionality.

### CODE REVIEW

**Comment by tVeratti** [comment]
Context menu makes me think right-click. Is the requirement on just a basic left click anywhere in the cell?

**Comment by mathom** [comment]
[100% (0.0%) vs master 100%](https://jenkins.topbox.io/job/Topbox/job/rci-components/job/PR-639/1/)

**Comment by troyblank** [comment]
@tVeratti  it's a left click... but yea context usually shows for right in an OS, but problem is there is no right click in browser and it pretty much is a context menu in my opinion.

**Comment by tVeratti** [comment]
Well we could hook into 'onContextMenu' and just block the default one from opening, thus creating our own right click menu.

File: components/table/cell.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        const { title, value, columnStyle } = this.props;
+
+        return (
+          <div
+            className={columnStyle}
+            onClick={this.onClick}
+          >
+            <div className={styles['cell__vertical-label']}>
+              { title && <span>{ title }</span> }
+            </div> { value }
```
Comment (tVeratti) [inline components/table/cell.js]: maybe next line the value? Looks kinda funky hanging on after the closed `</div>`

Comment (troyblank) [reply components/table/cell.js]: Sure! I adjusted this.

**Comment by troyblank** [comment]
@tVeratti  I noticed a small issue I introduce, please check last commit and re-review pretty please.

**Comment by mathom** [comment]
[100% (0.0%) vs master 100%](https://jenkins.topbox.io/job/Topbox/job/rci-components/job/PR-639/2/)

File: components/table/row.js [hunk 2]
```diff
@@ -48,17 +36,24 @@ export default class Row extends PureComponent {
...
-        const { titles, row, link, cellStyles } = this.props;
+        const { titles, row, link, tableHasLinks, cellStyles } = this.props;
 
         return (
           <li
             className={styles.row}
             onDoubleClick={this.navigateToLink}
           >
-            { link && renderLink(link) }
+            { tableHasLinks && renderLink(link) }
```
Comment (tVeratti) [inline components/table/row.js]: Good catch.

**Pull request merged**